### PR TITLE
Make triage color-code clusters to show fuzziness.

### PR DESCRIPTION
**Pull request by rmmh**
[staging](https://storage.googleapis.com/k8s-gubernator/triage/staging/index.html)

Also includes some other minor fixes.

### CODE REVIEW

File: triage/summarize.py [hunk 1]
```diff
@@ -290,11 +287,53 @@ def tests_group_by_job(tests, builds):
                   key=lambda (k, v): (-len(v), k))
 
 
+SPAN_RE = re.compile(r'\w+|\W+')
+
+def common_spans(xs):
+    '''
```
Comment (fejta) [inline triage/summarize.py]: `"""this"""`  and `'that'`

File: triage/summarize.py [hunk 2]
```diff
@@ -290,11 +287,53 @@ def tests_group_by_job(tests, builds):
...
+    Finds something similar to the longest common subsequence of xs, but much faster.
+
+    Returns a list of [matchlen_1, mismatchlen_2, matchlen_2, mismatchlen_2, ...], representing
+    sequences of the first element of the list that are present in all members.
+    '''
+    common = None
+    for x in xs:
+        x_split = SPAN_RE.findall(x)
+        if common is None:
```
Comment (fejta) [inline triage/summarize.py]: Consider noting that we want to distinguish first iteration from an empty intersection

File: triage/summarize.py [hunk 3]
```diff
@@ -290,11 +287,53 @@ def tests_group_by_job(tests, builds):
...
+            common = set(x_split)
+        else:
+            common.intersection_update(x_split)
```
Comment (fejta) [inline triage/summarize.py]: You don't want to use &=?

File: triage/render.js [hunk 4]
```diff
@@ -220,9 +220,26 @@ function renderLatest(el, keyId) {
...
+  if (!spans) {
+    return [text];
+  }
+  var out = [];
+  var c = 0;
+  for (var i = 0; i < spans.length; i += 2) {
+    out.push(text.slice(c, c + spans[i]));
+    c += spans[i];
+    if (i + 1 < spans.length) {
+      out.push(createElement('span', {className: 'mm'}, text.slice(c, c + spans[i+1])));
```
Comment (fejta) [inline triage/render.js]: Give it a tooltip to note that this text differs between instances?

File: triage/render.js [hunk 5]
```diff
@@ -220,9 +220,26 @@ function renderLatest(el, keyId) {
   }
 }
 
+function renderSpans(text, spans) {
```
Comment (fejta) [inline triage/render.js]: doc?

File: triage/render.js [hunk 6]
```diff
@@ -220,9 +220,26 @@ function renderLatest(el, keyId) {
...
+  }
+  return out;
+}
+
 // Render a section for each cluster, including the text, a graph, and expandable sections
 // to dive into failures for each test or job.
-function renderCluster(top, key, keyId, text, tests) {
+function renderCluster(top, key, keyId, text, tests, spans) {
   function plural(count, word, suffix) {
     return count == 1 ? count + ' ' + word : count + ' ' + word + suffix;
```
Comment (fejta) [inline triage/render.js]:
```
prefix = regular_if_statement ? "would " + "be " + "easier " : "is " + "harder ";
return "this" + prefix + " to" + " read";
```

File: triage/render.js [hunk 7]
```diff
@@ -239,9 +256,9 @@ function renderCluster(top, key, keyId, text, tests) {
     createElement('h2', null, [
       `${plural(clusterSum, 'FAILURE', 'S')} (${todayCount} TODAY) MATCHING `,
       createElement('a', {href: '#' + keyId}, keyId),
-      createElement('a', {href: 'https://github.com/search?type=Issues&q=' + keyId, target: '_blank', rel: 'noopener'}, 'github search')
+      createElement('a', {href: 'https://github.com/search?type=Issues&q=user:kubernetes%20' + keyId, target: '_blank', rel: 'noopener'}, 'github search')
```
Comment (fejta) [inline triage/render.js]: I would expect this to be `org:kubernetes`, why user?

File: triage/summarize.py (hunk 1 above)
Comment (rmmh) [inline triage/summarize.py]: done

File: triage/summarize.py (hunk 2 above)
Comment (rmmh) [inline triage/summarize.py]: done

File: triage/summarize.py (hunk 3 above)
Comment (rmmh) [inline triage/summarize.py]: set &= list doesn't work, but intersection_update does

File: triage/render.js (hunk 5 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 4 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 6 above)
Comment (rmmh) [inline triage/render.js]: done

File: triage/render.js (hunk 7 above)
Comment (rmmh) [inline triage/render.js]: they're apparently synonyms. changed to `org:kubernetes`

**Pull request merged**